src/uct/ib/ud/base/ud_iface.h

```diff
@@ -257,10 +294,25 @@ static UCS_F_ALWAYS_INLINE void uct_ud_leave(uct_ud_iface_t *iface)
     UCS_ASYNC_UNBLOCK(iface->super.super.worker->async);
 }
 
+static UCS_F_ALWAYS_INLINE unsigned
+uct_ud_grh_get_dgid_len(struct ibv_grh *grh)
+{
+    static const uint8_t ipmask = 0xf0;
+    uint8_t ipver               = ((*(uint8_t*)grh) & ipmask);
+
+    return (ipver == (6 << 4)) ? UCS_IPV6_ADDR_LEN : UCS_IPV4_ADDR_LEN;
+}
+
+union ibv_gid* uct_ud_grh_get_dgid(struct ibv_grh *grh, size_t dgid_len);
+
 static UCS_F_ALWAYS_INLINE int
-uct_ud_iface_check_grh(uct_ud_iface_t *iface, void *grh_end, int is_grh_present)
+uct_ud_iface_check_grh(uct_ud_iface_t *iface, void *packet, int is_grh_present)
```

Comment:
1. need 2 space lines in H file
2. pls move the prototype of uct_ud_grh_get_dgid to line 267, along with other function prototypes

Reply:
everything else looks good besides this comment